django/contrib/messages/storage/cookie.py

```diff
@@ -166,17 +168,23 @@ def _decode(self, data):
         if not data:
             return None
         try:
-            decoded = self.signer.unsign(data)
+            return self.signer.unsign_object(data, serializer=MessageSerializer)
+        # RemovedInDjango40Warning: when the deprecation ends, replace both with:
+        #
+        # except Exception:
```

Comment:
Why we want to catch all exceptions? I would catch only `(signing.BadSignature, json.JSONDecodeError)`.

Reply:
Fair, I didn't really think about which exceptions could all happen :)